R/ssm_to_proteinpaint.R

```diff
@@ -0,0 +1,142 @@
+#' checking purposes only. Setting this to TRUE, does not produce an output compatible with 
+#' Protein paint. The default is FALSE.
+#'
+#' @return A data frame.
+#' 
+#' @import tidyr
+#' @export
+#'
+#' @examples
+#' library(GAMBLR.data)
+#' 
+#' # get maf data frame
+#' my_maf = get_coding_ssm(basic_columns = FALSE)
+#' 
+#' # convert maf to ProteinPaint format
+#' pp_df = ssm_to_proteinpaint(my_maf)
+#' 
+ssm_to_proteinpaint = function(maf_data,
+                               return_removed_rows = FALSE){
+  
+  # check for required columns in maf_data 
+  maf_req_cols = c("Hugo_Symbol", "RefSeq", "Chromosome", "Start_Position", "HGVSp_Short", 
+                    "Variant_Classification")
+  maf_req_cols_not_present = maf_req_cols %>% 
+    "["(! . %in% names(maf_data))
+  if( length(maf_req_cols_not_present) > 0 ){
+    k = paste(maf_req_cols_not_present, collapse = ", ") %>% 
+      gettextf("Required columns missing in the input MAF data frame: %s.", .)
+    stop(k)
+  }
+  
+  # check for optional columns in maf_data
+  maf_opt_cols = c("Mutation_Status", "t_alt_count", "t_depth", "n_alt_count", "n_depth", 
+                   "Tumor_Sample_Barcode", "Reference_Allele")
+  maf_opt_cols_not_present = maf_opt_cols %>% 
+    "["(! . %in% names(maf_data))
+  if( length(maf_opt_cols_not_present) > 0 ){
+    k = paste(maf_opt_cols_not_present, collapse = ", ") %>% 
+      gettextf("Warning: Otional columns missing in the input MAF data frame: %s.", .)
+    message(k)
+  }
+  
+  # add metadata columns to maf_data
+  metadata = get_gambl_metadata(seq_type_filter = c("genome", "capture", "mrna"))
+  maf_data = left_join(maf_data, metadata, by = join_by(Tumor_Sample_Barcode == sample_id))
+  
+  # check for optional columns in maf_data, including columns from the metadata
+  maf_opt_cols = c("pathology", "patient_id", "time_point")
+  maf_opt_cols_not_present = maf_opt_cols %>% 
+    "["(! . %in% names(maf_data))
+  if( length(maf_opt_cols_not_present) > 0 ){
+    k = paste(maf_opt_cols_not_present, collapse = ", ") %>% 
+      gettextf("Warning: Otional columns missing in the metadata (created internally): %s.", .)
+    message(k)
+  }
+  
+  # adequate maf_data to ProteinPaint visualization format
+  maf_data = rename(maf_data, gene = Hugo_Symbol, chromosome = Chromosome, 
+                    start = Start_Position, class = Variant_Classification,
+                    refseq = RefSeq, aachange = HGVSp_Short, origin = Mutation_Status,
+                    disease = pathology, mutant_in_tumor = t_alt_count, 
+                    total_in_tumor = t_depth, mutant_in_normal = n_alt_count, 
+                    total_in_normal = n_depth, patient = patient_id, 
+                    sample = Tumor_Sample_Barcode, sampletype = time_point,
+                    REF = Reference_Allele) %>% 
+    mutate(class = recode(class,
+                          Missense_Mutation = "missense",
```

Comment:
I trust you checked this with a scenario when the full maf is supplied, not previously subsetted to only coding mutations, and such input maf_data works perfectly here so no filtering is required here :smile:

Reply:
Thank you very much for your comment. I realized an error that was removing non-coding mutations. I fixed it and also added the new parameter `coding_only`.